src/Bicep.LangServer/Handlers/BicepDecompileCommandHandler.cs

```diff
@@ -0,0 +1,191 @@
+        public string? mainBicepPath; // non-null if errorMessage==null
+        public DecompiledFile[] outputFiles;
+        public string[] conflictingOutputPaths; // client should verify overwrite with user
+
+        private BicepDecompileCommandResult(
+            string output,
+            string? errorMessage,
+            string? mainBicepPath,
+            DecompiledFile[] outputFiles,
+            string[] conflictingOutputPaths)
+        {
+            this.decompileId = Guid.NewGuid().ToString();
+            this.errorMessage = errorMessage;
+            this.output = output;
+            this.mainBicepPath = mainBicepPath;
+            this.outputFiles = outputFiles;
+            this.conflictingOutputPaths = conflictingOutputPaths;
+        }
+
+        public BicepDecompileCommandResult(
+            string output,
+            string? errorMessage
+        ) : this(output, errorMessage, null, new DecompiledFile[] { }, new string[] { })
+        {
+        }
+
+        public BicepDecompileCommandResult(
+            string output,
+            string mainBicepPath,
+            DecompiledFile[] outputFiles,
+            string[] conflictingOutputPaths
+        ) : this(output, null, mainBicepPath, outputFiles, conflictingOutputPaths)
+        {
+        }
+    }
+
+    /// <summary>
+    /// Handles a request from the client to decompile a JSON file for given a file path, creating one or more bicep files
+    /// </summary>
+    public class BicepDecompileCommandHandler : ExecuteTypedResponseCommandHandlerBase<BicepDecompileCommandParams, BicepDecompileCommandResult>
+    {
+        private readonly TemplateDecompiler templateDecompiler;
+        private readonly TelemetryAndErrorHandlingHelper<BicepDecompileCommandResult> telemetryHelper;
+
+        public BicepDecompileCommandHandler(
+            ISerializer serializer,
+            IFeatureProviderFactory featureProviderFactory,
+            INamespaceProvider namespaceProvider,
+            IModuleRegistryProvider registryProvider,
+            ILanguageServerFacade server,
+            ITelemetryProvider telemetryProvider,
+            IApiVersionProviderFactory apiVersionProviderFactory,
+            IBicepAnalyzer bicepAnalyzer,
+            IFileResolver fileResolver)
+            : base(LangServerConstants.DecompileCommand, serializer)
+        {
+            this.telemetryHelper = new TelemetryAndErrorHandlingHelper<BicepDecompileCommandResult>(server.Window, telemetryProvider);
+            this.templateDecompiler = new TemplateDecompiler(featureProviderFactory, namespaceProvider, fileResolver, registryProvider, apiVersionProviderFactory, bicepAnalyzer);
+        }
+
+        public override Task<BicepDecompileCommandResult> Handle(BicepDecompileCommandParams parameters, CancellationToken cancellationToken)
+        {
+            return telemetryHelper.ExecuteWithTelemetryAndErrorHandling(() =>
+            {
+                if (parameters.jsonUri is null)
```

Comment:
Is this check required? DocumentUri in BicepDecompileCommandParams is non null. Correct me if I am wrong.

Reply:
removed